ci/cmake.yml

```diff
@@ -0,0 +1,17 @@
+name: CMake CI
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v1
+    - name: configure
+      run: cmake .
```

Comment:
You should rather avoid in-source builds. Some projects won't even allow it.

Reply:
While I agree with the sentiment, determining and creating a build directory in a cross platform way seems problematic.  I don't want to have the starter workflow have a bunch of per-platform logic to calculate a suitable build directory name, since I'd then have to worry about the chosen directory name conflicting with an existing directory from the checkout.  

If you have any suggestions on how to resolve this in a platform-neutral way that doesn't end up obscuring the cmake specific content, I'd be happy to incorporate them.

Reply:
Isn't it pretty common to build in a "build" subdirectory of the source? I've seen that quite a bit. I've used that myself, but I've also used the case where the build directory is at the same level as the source file. Consider the project `foo`:

````
foo
|
|-foo-src
  |-CMakeLists.txt
|-foo-build
|-foo-build-special-version
````

I think these two cases cover the majority of cases in how CMake is typically configured/used. Have you observed any other patterns?

Reply:
I've discovered the `cmake -E make_directory` command provides a cross platform way to make directories.  I will update this PR to reflect the common build practice.

Reply:
Can't you just specify the build directory at configuration time? 

````cmake
cmake -S <source> -B <build> ...<other flags>
````
This should create the build directory if it doesn't exist. Or am I wrong?

Reply:
The `-S <source> -B <build>` syntax doesn't work with CMake 3.12, which is what's on at least some of the build hosts.  However, the make directory mechanism will work.

Reply:
Ah I see, didn't realize we're working with CMake 3.12.

Reply:
Me either, which is why I nearly lost my mind trying to debug why the -S / -B stuff wasn't working.

Reply:
Thanks for handling it though, looking forward to using this action for my cmake projects!

Reply:
https://commons.wikimedia.org/wiki/File:Cmake.svg